### PR TITLE
automate updating of requirements.txt

### PR DESCRIPTION
**Pull request by StavisT**
Adds a script to simplify the update of requirements.txt files

This is a suggestion for making it easier for users of the template to update requirements.txt files. This is something I have found useful and therefore wanted to add it to the template so others can benefit from this.

Let me know if you have any suggestions for further improving this.

Timeline: not urgent

### CODE REVIEW

**Comment by ks93** [comment]
This assumes all functions has all the same dependencies, right? Not sure how often that's the case. I guess the deployment time and resource requirements increases with additional requirements, so would likely just want to install necessary deps per function

**Comment by StavisT** [comment]
> This assumes all functions has all the same dependencies, right? Not sure how often that's the case. I guess the deployment time and resource requirements increases with additional requirements, so would likely just want to install necessary deps per function

yes it does assume that.

How would you handle different dependencies across different functions? @ks93 

For very simple functions, e.g. only using the cognite-sdk and pandas, I agree that this would not be necessary. in my case the overlap of requirements is so large that I am willing to wait a little extra for the build instead of updating the requirements.txt files on a case by case.

**Comment by ks93** [comment]
> 

Found it!: https://cognitedata.slack.com/archives/CUJCC1S5U/p1618393315002900

**Comment by scottmelhop** [comment]
You should aim for your dependencies to be as lean as possible. Given that there are multiple functions defined in one repo each function should define its dependencies. Understand this script may be a convenience however it encourages bad practice

**Comment by ks93** [comment]
@scottmelhop @StavisT Could poetry dependency groups be an idea? We actually discussed that concept in the thread I linked above, but it didn't exist in poetry at the time. 

Seems to let you install like this:
`poetry add cognite-sdk --group func1`
which can be installed with
`poetry install --with func1`

Doesn't seem like it supports `poetry export --with func1` though

 https://python-poetry.org/docs/master/managing-dependencies/#dependency-groups

**Comment by scottmelhop** [comment]
You DS love to add complexity

**Comment by haakonvt** [comment]
I am happy to accept this _purely_ as part of the readme. What do you think?

**Comment by StavisT** [comment]
The intention here is not to add complexity, so if you think that is the case then we can close this PR.

The background for this addition is purely pragmatic and to ensure that cognite functions does not fail due to issues with dependencies or breaking changes, because during development you will (most likely) run the code with your local environment and expect it to run in the same way in the cognite function.

I agree that cases where there is a very simple cognite function with limited dependencies, this shell script is not needed because you can simply keep track of these dependencies, then this directory can be excluded from the for loop in the shell script.
In cases where you have two or more cognite functions in the same repo and you dont want to keep track of the dependencies independently I find this shell script useful.

**Comment by jaketbouma** [comment]
Tend to agree that this should be in README or in a reference where you can pick snippets from.
Adding a script like this as part of the repo is just more stuff to maintain.

File: update_requirements.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+for FUNCTION in example_function1 example_function2
+do
+	poetry export -f requirements.txt --without-hashes > $FUNCTION/requirements.txt
```
Comment (jaketbouma) [inline update_requirements.sh:5]: Adding this line in the README along with some words to describe the workflow to update would be enough, for now.

**Comment by haakonvt** [comment]
I'm closing this, but feel free to reopen if you want to address the comments 😄

**Pull request closed without merging**